### ISSUE TITLE
Feature column support.

### ISSUE THREAD

**Issue by yaochitc** [tensorflow/probability#225]
Currently it is not easy to use tfp with features in different types(categorical, interactions, etc.), is it suitable to implement something like feature column in tensorflow that works with probabilistic weights?

**Comment by brianwa84** [comment]
Can you add a little more detail?

I guess here you want something like tfp.layers.Embedding[Flipout|Reparam|LocalReparam]?
Essentially, the marriage of tf.keras.layers.Embedding with tfp.layers.DenseFlipout (or DenseReparameterization etc.)?

**Comment by yaochitc** [comment]
No, as the examples show, currently when I need to create models like hlm with categorical variables or interactions, I have to do the feature transformation by myself. I just wondered if something like feature columns with probabilistic weights can be added to make it easy to use.
